src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.cs

```diff
@@ -257,6 +257,44 @@ async Task Test(bool expectOutput)
             }
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNetCore))]
+        public async Task ListOfTupleTest()
+        {
+            string source = """
+                using Microsoft.Extensions.Configuration;
+                using System;
+                using System.Collections.Generic;
+
+                public class Program
+                {
+                    public static void Main()
+                    {
+                        ConfigurationBuilder configurationBuilder = new();
+                        IConfiguration config = configurationBuilder.Build();
+
+                        var settingsSection = config.GetSection("Settings");
+
+                        Settings options = settingsSection.Get<Settings>()!;
+                    }
+                }
+
+                public class Settings
+                {
+                    public List<(string Item1, string? Item2)>? Items { get; set; }
+                    // public List<string>? Items { get; set; }
+                }
+                """;
+
+            ConfigBindingGenRunResult result = await RunGeneratorAndUpdateCompilation(source, assemblyReferences: GetAssemblyRefsWithAdditional(typeof(ConfigurationBuilder), typeof(List<>)));
+            Assert.NotNull(result.GeneratedSource);
+            Assert.Empty(result.Diagnostics);
+
+            // Ensure the generated code can be compiled.
+            // If there is any compilation error, exception will be thrown with the list of the errors in the exception message.
+            byte[] emittedAssemblyImage = CreateAssemblyImage(result.OutputCompilation);
```

Comment:
Interesting this is the first time we're compiling the generator output -- we didn't have any other patterns to follow here for validating this?

Reply:
We do in other places, for example:

https://github.com/dotnet/runtime/blob/27604b57bd2e9c9aa46d005db7c4e387d461b5b6/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorParserTests.cs#L1029

used to compile and run the generated code. In my case I wanted to validate it compile fine. This is used also in other generators like regex and json.